tests/unit/scheduling_group_nesting_test.cc

```diff
@@ -94,43 +95,58 @@ SEASTAR_TEST_CASE(test_basic_ops) {
 static future<> run_busyloops(std::vector<seastar::scheduling_group> groups, std::vector<float> expected) {
     std::vector<uint64_t> counts;
     std::vector<future<>> f;
-    auto run_and_count = [&groups] (uint64_t& c) -> future<> {
-        unsigned total_run_ms = 100 * groups.size();
-        auto now = std::chrono::steady_clock::now();
-        auto start_count = now + std::chrono::milliseconds(20);
-        auto stop_count = now + std::chrono::milliseconds(total_run_ms - 20);
-        auto stop = now + std::chrono::milliseconds(total_run_ms);
+    enum class loop_stage { warmup, count, stop };
+    loop_stage stage = loop_stage::warmup;
+
+    auto run_and_count = [&stage] (uint64_t& c) -> future<> {
         do {
             auto stop = std::chrono::steady_clock::now() + std::chrono::microseconds(10);
             while (std::chrono::steady_clock::now() < stop) ;
 
-            auto now = std::chrono::steady_clock::now();
-            if (now >= start_count && now < stop_count) {
+            if (stage == loop_stage::count) {
                 c++;
             }
             co_await yield();
-        } while (std::chrono::steady_clock::now() < stop);
+        } while (stage != loop_stage::stop);
     };
     counts.reserve(groups.size());
     f.reserve(groups.size());
     for (auto& g : groups) {
         counts.push_back(0);
         f.push_back(with_scheduling_group(g, [&c = counts.back(), &run_and_count] { return run_and_count(c); }));
     }
-    co_await when_all(f.begin(), f.end());
+
+    fmt::print("Warmup for 20ms\n");
+    co_await seastar::sleep(std::chrono::milliseconds(20));
+
+    stage = loop_stage::count;
+    auto pause = std::chrono::milliseconds(100);
```

Comment:
This variable had me puzzled. We're not pausing for 100ms, we're scheduling a sample in 100ms.

Suggest to rename to sample_period.